bdk_core/src/spk_tracker.rs

```diff
@@ -116,19 +98,20 @@ impl<I: Clone + Ord> SpkTracker<I> {
     pub fn iter_txout_full<'a>(
         &'a self,
         chain: &'a SparseChain,
+        graph: &'a TxGraph,
     ) -> impl DoubleEndedIterator<Item = (I, FullTxOut)> + 'a {
         self.txouts.iter().filter_map(|(outpoint, spk_index)| {
-            Some((spk_index.clone(), chain.full_txout(*outpoint)?))
+            Some((spk_index.clone(), chain.full_txout(graph, *outpoint)?))
         })
     }
 
     pub fn iter_txout<'a>(
         &'a self,
-        chain: &'a SparseChain,
+        graph: &'a TxGraph,
     ) -> impl DoubleEndedIterator<Item = (I, OutPoint)> + 'a {
         self.txouts
             .iter()
-            .filter(|(outpoint, _)| chain.get_tx(outpoint.txid).is_some())
+            .filter(|(outpoint, _)| graph.tx(&outpoint.txid).is_some())
```

Comment:
I don't think this makes sense (it might not have made sense before). What this function used to do was to iterate over txouts that were in an actual chain. Now it iterates over txouts that have been added to `TxGraph`. Maybe change to `iter_txout_in_chain` and change the argument back to `SparseChain`.

Reply:
Does mempool txouts go here as well?